text/0000-location.md

```diff
@@ -0,0 +1,409 @@
+- LocationDelete - remove a Location from state.
+ 
+_*Disclaimer: LocationDelete is a potentially hazardous operation and needs to 
+be done with care._
+
+
+## Permissions
+[permissions]: #permissions
+
+Creation of GS1 locations is restricted to agents acting on behalf of the 
+organization that is associated with the GS1 company prefix encoded in the GLN. 
+An organization must have the GS1 company prefix in the �gs1_company_prefixes� 
+metadata field for its Pike organization. (Organization-level metadata will need
+to be implemented in Pike.) When a location is created, its owning organization 
+is stored with the location in an �owner� field.
+
+Updates to locations are restricted to agents acting on behalf of the 
+organization stored in the location�s owner field. Only property fields can be 
+updated. They are updated by providing a complete and updated list of the 
+properties, and overwrites the entire list of property values. Location_id, 
+location_namespace, and owner fields are immutable.
+
+Deletion of locations is restricted to agents acting on behalf of the 
+organization in the location�s owner field. A setting will turn off deletion 
+entirely, with the intent that it could be enabled/disabled by a Grid 
+administrator. Disabling delete is useful because external systems may have 
+references to these locations and deleting them could leave dangling references.
+
+
+## GLN Validation
+[gln validation]: #glnvalidation
+
+Creation of a reusable GLN validation function to programmatically express the 
+equation used to validate a GLN. It validates GLN format to avoid mistype errors
+similar to a credit card validation. It's implemented as an extensible function 
+such that further validation steps can be added, if needed. For details on the 
+equation see: [Check digit validation](https://www.gs1us.org/tools/check-digit-calculator "Check digit validation")
+
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## State
+[state]: #state
+
+### Location Representation
+
+The primary object stored in state is �Location�, which consists of a 
+location_id (GS1 GLN), an owner (org_id compatible with Pike), and a list of 
+property name/value pairs. The properties available are defined by the Grid 
+Location Schema Transaction Family and are restricted to the fields and rules of
+the GS1 Location Property Schema. Transactions which are responsible for setting
+location state values must ensure that the properties conform with the 
+requirements of the GS1 Location Property Schema.
+
+The GS1 GLN is an identifier (location_id) used to identify entities or 
+locations. A GLN is the data transmitted from a data carrier (barcode, rfid, 
+etc.) scan and is made up of a company prefix, location reference, and check 
+digit.
+
+### Referencing Locations
+
+Locations are uniquely referenced by their location_id and location_namespace.
```

Comment:
If we add in the above protobuf message, change location_namespace to location_type here.